File: app/src/main/java/com/ishow/noah/modules/sample/pulltorefresh/SamplePullToRefreshPresenter.java
package com.ishow.noah.modules.sample.pulltorefresh;

import android.content.Context;
import android.support.annotation.NonNull;

import com.ishow.common.utils.http.rest.Http;
import com.ishow.common.utils.http.rest.HttpError;
import com.ishow.noah.modules.sample.entries.page.PageJob;
import com.ishow.noah.utils.http.AppHttpCallBack;

class SamplePullToRefreshPresenter implements SamplePullToRefreshContract.Presenter {

    private SamplePullToRefreshContract.View mView;

    SamplePullToRefreshPresenter(SamplePullToRefreshContract.View view) {
        mView = view;
    }


    @Override
    public void getList(Context context, final int pageNumber, final boolean loadingView) {
        if (loadingView) mView.showLoading(null, false);
        String url = "http://xiaomidagong.com/Home/SelectFactoryInfoType?title=&page=" + String.valueOf(pageNumber) + "&pagesize=10&type=1";
        Http.post()
                .url(url)
                .execute(new AppHttpCallBack<PageJob>(context) {
                    @Override
                    protected void onFailed(@NonNull HttpError error) {
                        if (loadingView) mView.showError(null, false, error.getCode());
                    }

                    @Override
                    protected void onSuccess(PageJob result) {
                        if (result == null || result.getResults().isEmpty()) {
                            if (loadingView) mView.showEmpty(null);
                        } else {
                            if (loadingView) mView.dismissLoading(false);
                            mView.updateView(result.getResults(), result.getTotalRows());
                        }
                    }
                });
    }
}

File: app/src/main/java/com/ishow/noah/modules/sample/pulltorefresh/SamplePullToRefreshPresenter.kt
package com.ishow.noah.modules.sample.pulltorefresh

import android.content.Context
import com.ishow.common.entries.status.Error
import com.ishow.common.entries.status.Loading

import com.ishow.common.utils.http.rest.Http
import com.ishow.common.utils.http.rest.HttpError
import com.ishow.noah.modules.sample.entries.page.PageJob
import com.ishow.noah.utils.http.AppHttpCallBack

internal class SamplePullToRefreshPresenter(private val mView: SamplePullToRefreshContract.View) :
    SamplePullToRefreshContract.Presenter {


    override fun getList(context: Context, pageNumber: Int, loadingView: Boolean) {
        if (loadingView) mView.showLoading(Loading.view())
        val url = "http://xiaomidagong.com/Home/SelectFactoryInfoType?title=&page=$pageNumber&pagesize=10"
        Http.post()
            .url(url)
            .execute(object : AppHttpCallBack<PageJob>(context) {
                override fun onFailed(error: HttpError) {
                    if (loadingView) mView.showError(Error.view(error.message))
                }

                override fun onSuccess(result: PageJob?) {
                    if (result == null || result.getResults().isEmpty()) {
                        if (loadingView) mView.showEmpty()
                    } else {
                        if (loadingView) mView.dismissLoading()
                        mView.updateView(result.getResults(), result.getTotalRows())
                    }
                }
            })
    }
}